### ISSUE TITLE
Regression release-2.30.6 with MSVC and BUILD_SHARED_LIBS=OFF

### ISSUE THREAD

**Issue by scribam** [libsdl-org/SDL#10462]
With tag `release-2.30.6` (and branch `release-2.30.x`), build SDL on MSVC with `-DBUILD_SHARED_LIBS=OFF`

I have this error:
```
--  Build Shared Library: OFF
--  Build Static Library: ON
--  Build Static Library with Position Independent Code: OFF
-- 
CMake Error at CMakeLists.txt:3530 (set_property):
  set_property could not find TARGET SDL2.  Perhaps it has not yet been
  created.
```
Related commit: https://github.com/libsdl-org/SDL/commit/5a072d38dbf63550bc243187694f77f270c79830
Author: @madebr 

With `-DBUILD_SHARED_LIBS=OFF`, the shared library is not built and the target `SDL2` can not exist

**Comment by scribam** [comment]
Potential fix:
```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 284c2a6f5..dfed28628 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3520,7 +3520,10 @@ if(SDL_TEST)
 endif()

 if(MSVC AND NOT SDL_LIBC)
-  set(targets SDL2)
+  set(targets "")
+  if(TARGET SDL2)
+    list(APPEND targets SDL2)
+  endif()
   if(TARGET SDL2-static)
     list(APPEND targets SDL2-static)
   endif()
```

**Comment by madebr** [comment]
Thanks. This was an unintential nudge to use `SDL2.dll`, and not use static libraries. :)

Fixed in SDL2 (7cf3234efeb7a68636bcfdfb3b1507b43fbb0845), and in in the 2.30.x (b7457ff61b0b520b9d94ef77fe6a410a6c8141f4).
